gssutils/csvw/mapping.py

```
@@ -256,6 +275,17 @@ def _as_csvw_object(self):
                         propertyUrl=URI("http://purl.org/linked-data/cube#measureType"),
                         valueUrl=URI(obj["measure"])
                     )
+                elif "datatype" in obj and not ("measure" in obj or "unit" in obj):
```

Since the code which deals with `measure` and `unit` is above here, I would argue that if either of them are in this object containing `datatype` then we'd probably want an exception here to tell us what's wrong.

I'm all for failing hard when we have an inconsistency.

---

Aye. This is one of a few different ways I could think of doing this and still feels a bit dodgy. Failing fast is good.

---

Agree with Rob (I like code that screams at me when things go wrong) but it feels like part of a/the future refactor. 

For me, every column "thing" should be mapped to a "type" up front, then errors are thrown for any and all oddities before we get on to actually doing anything (or similar, I'm sure everyone has their own thoughts). Validation of input I guess.